backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/JsonApi.java

```diff
@@ -37,9 +37,16 @@ public static void getJsonReport(Context ctx) throws JsonProcessingException {
     Map<String, Object> jsonResponse = new HashMap<>();
 
     for (String reportId : reports.split(",")) {
-      Map<String, Object> reportObject = reportList.get(Integer.parseInt(reportId.trim()));
+
+      Map<String, Object> reportObject =
+          reportList.stream()
+              .filter(r -> reportId.equals(r.get("id")))
+              .findFirst()
+              .orElseGet(() -> null);
+      if (reportObject == null) {
+        reportObject = reportList.get(Integer.parseInt(reportId.trim()));
+      }
```

Comment:
Feels a bit hacky but it works

Reply:
indeed. I would be happyier to discontinue this feature (so not allow retrieve by index). Please advice